functions/payment.js

```diff
@@ -125,6 +140,52 @@ exports.createPaymentIntent = functions.https.onCall(async (data, context) => {
   return { client_secret: paymentIntent.client_secret };
 });
 
+exports.confirmPaymentIntent = functions.https.onCall(async (data, context) => {
+  if (!context.auth || !context.auth.token) {
+    throw new functions.https.HttpsError("unauthenticated", "Please log in");
+  }
+
+  if (context.auth.token.aud !== PROJECT_ID) {
+    throw new functions.https.HttpsError("permission-denied", "Token invalid");
+  }
+
+  if (
+    !(
+      data &&
+      data.venueId &&
+      data.eventId &&
+      data.paymentIntentId &&
+      context.auth.token.email &&
+      context.auth.token.user_id
+    )
+  ) {
+    throw new functions.https.HttpsError(
+      "invalid-argument",
+      "venue, event or return url data missing"
+    );
+  }
+
+  const documentToUpdate = (
+    await admin
+      .firestore()
+      .collection(PURCHASE_TABLE)
+      .doc(data.paymentIntentId)
+      .get()
+  ).data();
+
+  if (!["COMPLETE", "FAILED"].includes(documentToUpdate.status)) {
+    await admin
+      .firestore()
+      .collection(PURCHASE_TABLE)
+      .doc(data.paymentIntentId)
+      .update({
+        status: "CONFIRMATION_FROM_STRIPE_PENDING",
```

Comment:
when do we set status to PENDING. What's the difference between PENDING and CONFIRMATION_FROM...?